### ISSUE TITLE
Add support for default values.

### ISSUE THREAD

**Issue by lost-illusi0n** [lost-illusi0n/lostorm#3]
This will have to wait until #2 is complete. #2 will include the refactor of the annotation processor module and will make this issue much easier to add.
Default values will be configured by adding a regular default value to the field.
Example
```kotlin
@EntityDataClass
data class Human(
    val name: String = "Bob"
)
```
The annotation processor should be able to recognize the default value and correctly add them to the Column generated.

**Comment by lost-illusi0n** [comment]
Actually, due to a limitation in the way default values are compiled, it would be way too much work to support this implementation of default values. A better and simpler approach would be to simply add a ``defaultValue`` field to the Columns annotation that is a a String. Several other ORMs take this approach and it seems to work quite well.

**Comment by lost-illusi0n** [comment]
Awesome @DeMack, I'll assign you and once #2 is merged I'll mention you 😄

**Comment by DeMack** [comment]
👍

**Comment by DeMack** [comment]
Are the `'` necessary here `@Columns(defaultValue = "'bob'")`? It seems counter-intuitive to require essentially double string connotation. Since this is an ORM, and assuming it's a requirement for PostgreSQL strings, should we not allow the user to use just a typical `String` (e.g. `"bob"`) and append them for `String` types?

**Comment by DeMack** [comment]
Sounds good! 😁

**Comment by lost-illusi0n** [comment]
Awesome, I pushed the rewrite to the development branch. This should be quite simple to add now.
Iirc I did plan to add default values earlier on however I'm not sure if the configuration for default values in the Column class is still correct. Feel free to change what's needed in the mapper and annotations module. In the annotation processor, you just need to add another type to the ColumnProperty sealed class and generate the initializer. e.g.
```kotlin
sealed class ColumnProperty {
    class DefaultProperty(columns: Columns): ColumnProperty() {
        override val initializer = generate()

        private fun generate(): String {
            //do what you must with the columns
        }
    }
}
```
Good luck! 😄

**Comment by DeMack** [comment]
Sounds good!

**Comment by lost-illusi0n** [comment]
On second thought, you'll probably need to require the kotlin metadata of the property as to correctly get the type of the variable.

**Comment by DeMack** [comment]
OK. I'll give that a look too.

**Comment by lost-illusi0n** [comment]
resolved by #10
